File: leetcode/src/main/java/array/UniqueSumZero.java
package array;

import org.junit.jupiter.api.Test;
import java.util.Arrays;

import static org.junit.jupiter.api.Assertions.*;

/*
 	https://leetcode.com/problems/find-n-unique-integers-sum-up-to-zero/

 	Tạo mảng n số nguyên khác nhau có tổng  = 0.

 	=> Gán n - 1 số (>=0) đầu tiên là sum, gán số cuối cùng -sum.
 */
public class UniqueSumZero {

	public int[] sumZero(int n) {
		int sum = 0;
		int[] result = new int[n];
		for (int i = 0; i < n - 1; i++) {
			result[i] = i + 1;
			sum += result[i];
		}
		result[n - 1] = -sum;
		return result;
	}

	@Test
	public void test() {
	    assertEquals(0, Arrays.stream(sumZero(5)).sum());
	}

}

File: py-algorithm/array/unique_sum_zero.py
import unittest

from typing import List

"""
    https://leetcode.com/problems/find-n-unique-integers-sum-up-to-zero/

 	Tạo mảng n số nguyên khác nhau có tổng = 0.

 	=> Gán n - 1 số (>= 1) đầu tiên là sum, gán số cuối cùng -sum.
"""
class Solution(unittest.TestCase):

    def sumZero(self, n: int) -> List[int]:
        result = []
        for i in range(1, n):
            result.append(i)
        csum = sum(result)
        result.append(-csum)
        return result

    def test(self):
        self.assertEqual(0, sum(self.sumZero(8)))


if __name__ == '__main__':
    unittest.main()
